src/main/java/org/killbill/billing/client/KillBillHttpClient.java

```diff
@@ -117,12 +118,21 @@ public KillBillHttpClient(final String kbServerUrl, final String username, final
         this.apiKey = apiKey;
         this.apiSecret = apiSecret;
 
-
         final AsyncHttpClientConfig.Builder cfg = new AsyncHttpClientConfig.Builder();
 
+        if(requestTimeout != null)
+        {
+            cfg.setRequestTimeout(requestTimeout);
+            requestTimeoutSec = (int) Math.ceil((double) requestTimeout / 1000);
```

Comment:
Changed. A note if less than 1000ms is passed in the timeout will be set to zero.